### ISSUE TITLE
New rule: warn about useless indexOf conditions

### ISSUE THREAD

**Issue by HolgerJeromin** [typescript-eslint/typescript-eslint#6126]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-includes

### Description

I activated `@typescript-eslint/prefer-includes` rule at my code base and found the useless check on the indexOf result.

The mentioned rule is analyzing this code anyway, but I have no idea if this buggy code is best reported by this rule, another rule or typescript itself.

### Fail

```typescript
if (myArray.indexOf(myItem) >= -1) {
   //
}
```

### Pass

```typescript
if (myArray.indexOf(myItem) > -1) {
   //
}
```

**Comment by bradzacher** [comment]
To clarify - you're asking for a rule that specifically reports if they use `indexOf` with `>= -1`?

**Comment by HolgerJeromin** [comment]
Exactly. Sorry for being not clear enough.

Oh, I just saw that this is probably an enhancement for existing rule which I just found.
https://typescript-eslint.io/rules/no-unnecessary-condition/

**Comment by JoshuaKGoldberg** [comment]
I like the idea of checking cases like `indexOf` and `findIndex` being checked against always incorrect cases... Though, I wonder if it should be its own rule separate from `no-unnecessary-condition`? `no-unnecessary-condition` right now is kind of pure type-system-specific rule. I wonder if adding in known built-in API cases will over-inflate it...

**Comment by bradzacher** [comment]
I would actually love it if TS had some more specific number (and other primitive) types.
I.e. imagine if you could specify `indexOf(...): -1..number` or something. Then it would be pretty trivial to validate the logical is useless.
Looks like there's https://github.com/microsoft/TypeScript/issues/15480 to track that sort of thing at the type-level.

**Comment by bradzacher** [comment]
Based on our current expectations for enhancement requests, I believe this request fails the "nicheness" test:
- This request has been open for 4 years and has netted a total of 1 non-team member interactions.
- This request has been open for 4 years and nobody has moved to implement it.
- The code in question is a fairly niche bug vector that unit tests would catch easily
- Agentic code review should easily be able to catch the bug in question

For these reasons, I believe we should move to reject and close this. Thoughts @typescript-eslint/triage-team?

**Comment by Josh-Cena** [comment]
Agreed, this shouldn't be a ts-eslint concern.
